core/dispatcher/src/main/scala/whisk/core/dispatcher/ActivationFeed.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.dispatcher
+
+import scala.concurrent.duration.FiniteDuration
+import scala.language.postfixOps
+import scala.util.Try
+
+import akka.actor.Actor
+import akka.actor.actorRef2Scala
+import whisk.common.Logging
+import whisk.common.TransactionId
+import whisk.core.connector.MessageConsumer
+
+object ActivationFeed {
+    sealed class ActivationNotification
+
+    /** Pulls new messages from the message bus. */
+    case class FillQueueWithMessages()
+
+    /** Indicates resources are available because transaction completed, may cause pipeline fill. */
+    case class ContainerReleased(tid: TransactionId) extends ActivationNotification
+
+    /** Indicate resources are available because transaction failed, may cause pipeline fill. */
+    case class FailedActivation(tid: TransactionId) extends ActivationNotification
+}
+
+/**
+ * This actor polls the message bus for new messages and dispatches them to the given
+ * handler. The actor tracks the number of messages dispatched and will not dispatch new
+ * messages until some number of them are acknowledged.
+ *
+ * This is used by the invoker to pull messages from the message bus and apply back pressure
+ * when the invoker does not have resources to complete processing messages (i.e., no containers
+ * are available to run new actions).
+ *
+ * When the invoker releases resources (by reclaiming containers) it will send a message
+ * to this actor which will then attempt to fill the pipeline with new messages.
+ *
+ * The actor tries to fill the pipeline with additional messages while the number
+ * of outstanding requests is below the pipeline fill threshold.
+ */
+@throws[IllegalArgumentException]
+protected class ActivationFeed(
+    logging: Logging,
+    consumer: MessageConsumer,
+    maxPipelineDepth: Int,
+    longpollDuration: FiniteDuration,
+    handler: (String, Array[Byte]) => Any)
+    extends Actor {
+    import ActivationFeed.ActivationNotification
+    import ActivationFeed.FillQueueWithMessages
+
+    require(maxPipelineDepth >= 2 && maxPipelineDepth % 2 == 0, "max pipeline depth must be even and >= 2")
```

Comment:
It doesn't - left over from a previous iteration - will remove it. The only constraint is the one below.